doc/adding_to_an_adapter.md

```diff
@@ -1,120 +1,237 @@
-  <ResourceKinds>
-    <ResourceKind key="my_adapter_instance" nameKey="4" type="7" >
-        <!-- ... -->
-    </ResourceKind>
-  </ResourceKinds>
-</AdapterKind>
-```
-
-Once an adapter instance is defined, any configuration fields (`ResourceIdentifiers`) and credentials (`CredentialKind`) will be prompted to the user when creating an account in VMware Aria Operations on the `Data Sources` &rarr; `Integrations` page (See [Adding a Configuration Field to an Adapter Instance](#adding-a-configuration-field-to-an-adapter-instance) and [Adding a Credential](#adding-a-credential).
-After the account has been created, configuration fields will be available in the input to the `collect`, `test_connection`, and `get_endpoint_urls` methods. (See [Creating an Adapter Instance](#creating-an-adapter-instance).)
+Once an adapter instance is defined, any configuration fields (`ResourceIdentifiers` element) and credentials (`CredentialKind` element)
+will be prompted to the user when creating an account in VMware Aria Operations on the `Data Sources` &rarr; `Integrations`
+page (See [Adding a Configuration Field to an Adapter Instance](#adding-a-configuration-field-to-an-adapter-instance)
+and [Adding a Credential](#adding-a-credential). After the account has been created, configuration fields will be
+available in the input to the `collect`, `test`, and `get_endpoint` methods.
+(See [Creating an Adapter Instance](#creating-an-adapter-instance).)
 
-## Adding a Configuration Field to an Adapter Instance
+## Adding a Configuration Field to an Adapter Instance in the Object Model
 Adapter instance _identifiers_ distinguish between adapter instances from the same adapter. They also allow for user configuration.
 
-```xml
-<AdapterKind xmlns="http://schemas.vmware.com/vcops/schema" key="my_adapter" nameKey="1" version="1">
-  <ResourceKinds>
-    <ResourceKind key="my_adapter_instance" nameKey="5" type="7" >
-        <ResourceIdentifier dispOrder="1" key="instance" nameKey="6" required="true" type="string" identType="1"/>
-        <ResourceIdentifier dispOrder="2" key="ssl_mode" nameKey="7" required="true" type="string" identType="2" enum="true">
-            <enum default="true" value="Disable" />
-            <enum default="false" value="Require" />
-        </ResourceIdentifier>
-        <ResourceIdentifier dispOrder="3" key="max_events" nameKey="8" required="false" type="integer" identType="2"/>
-    </ResourceKind>
-  </ResourceKinds>
-  <!-- ... -->
-</AdapterKind>
-```
+> ### VMware Aria Operations Integration SDK Library
+> ```python
+> def get_adapter_definition() -> AdapterDefinition:
+>     definition = AdapterDefinition("MyAdapter", "My Adapter")
+>
+>     definition.define_string_parameter(
+>         "instance",
+>         label="Instance",
+>         required=True,
+>     )
+>     definition.define_enum_parameter(
+>         "ssl_mode",
+>         values= ["Disable"],
```

Comment:
I'd suggest we should always add all the enum values here, rather than relying on the `default` value being automatically added:
```suggestion
>         values=["Disable", "Require"],
```